### PR TITLE
Switch to Gold, added list_components workflow

### PR DESCRIPTION
**Pull request by mmzakhar**

### CODE REVIEW

File: scripts/build_linux.sh [hunk 1]
```diff
@@ -20,7 +22,8 @@ fortran)
   make && make run
   ;;
 dpc++)
-  source /opt/intel/oneapi/tbb/2021.1-beta10/env/vars.sh
+# shellcheck source=/dev/null
```
Comment (mmzakhar) [inline scripts/build_linux.sh:25]: Shellcheck flags sourcing of non-constant paths. This comment disables this check for the next line.

File: scripts/build_linux.sh [hunk 2]
```diff
@@ -5,10 +5,12 @@
 # SPDX-License-Identifier: MIT
 
 LANGUAGE=$1
+ONEAPI_RELEASE=$2
 
 git clone --depth 1 https://github.com/oneapi-src/oneAPI-samples.git
 
-source /opt/intel/oneapi/compiler/2021.1-beta10/env/vars.sh
+# shellcheck source=/dev/null
```
Comment (rscohn2) [inline scripts/build_linux.sh:12]: What is this for?

File: scripts/build_linux.sh (hunk 1 above)
Comment (rscohn2) [inline scripts/build_linux.sh:25]: and this?

Comment (mmzakhar) [reply scripts/build_linux.sh:25]: Shellcheck flags sourcing of non-constant paths. This comment disables this check for the next line.

**Review by rscohn2 — APPROVED** [review]

**Pull request merged**